File: ActivEYE-Service/src/main/java/cz/muni/fi/pa165/service/ActivityService.java
package cz.muni.fi.pa165.service;

import cz.muni.fi.pa165.activeye.entities.Activity;

import java.util.List;

/**
 * @author  Matej Vasko 422713
 * @version 1.0
 * @since   20-11-2016
 */
public interface ActivityService {

    /**
     * Creates a new Activity
     * @param a Activity to be created
     * @return void
     */
    void create(Activity a);

    /**
     * Updates an Activity
     * @param a Activity to be updated
     * @return void
     */
    void update(Activity a);

    /**
     * Deletes an Activity
     * @param a Activity to be deleted
     * @return void
     */
    void delete(Activity a);

    /**
     * Returns Activity with given id
     * @param id id of required Activity
     * @return Activity with given id
     */
    Activity findById(Long id);

    /**
     * Returns Activity with given name
     * @param name name of required Activity
     * @return Activity with given name
     */
    Activity findByName(String name);

    /**
     * Returns List of all Activities
     * @return List of all Activities
     */
    List<Activity> findAll();

}

File: ActivEYE-Webapp/src/main/webapp/js/service/ActivityService.js
/**
 * @author  Matej Vasko 422713
 * @version 1.0
 * @since   15-12-2016
 */
angular.module('activeye.services').factory('ActivityService', ActivityService);

ActivityService.$inject = ['$http'];

function ActivityService($http) {
    var vm = this;
    vm.create = function () {
        return $http.get('/pa165/rest/activities/create')
    };
    vm.update = function () {
        return $http.put('/pa165/rest/activities/update')
    };
    vm.delete = function () {
        return $http.delete('/pa165/rest/activities/delete/:activityId')
    };
    vm.findById = function () {
        return $http.get('/pa165/rest/activities/get/:activityId')
    };
    vm.findByName = function (name){
        return $http.get('/pa165/rest/activities/get',{
            params: {name: name}
        });
    };
    vm.getAll = function () {
        return $http.get('/pa165/rest/activities/get/all')
    };
    return vm;
}